src/github.com/getlantern/yamlconf/yamlconf.go

```diff
@@ -241,17 +245,25 @@ func (m *Manager) processUpdates() {
 
 func (m *Manager) processCustomPolling() {
 	for {
-		mutate, waitTime, err := m.CustomPoll(m.getCfg())
+		_, waitTime, _ := m.poll()
+		time.Sleep(waitTime)
+	}
+}
+
+func (m *Manager) poll() (func(cfg Config) error, time.Duration, error) {
+	m.pollMutex.Lock()
+	defer m.pollMutex.Unlock()
```

Comment:
As said, I think concurrency should have been prevented by the caller, so these locks shouldn't be necessary.